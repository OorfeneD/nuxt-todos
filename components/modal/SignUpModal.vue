<template>
  <div>
    <form class="p-4 rounded-xl shadow-md my-5 border border-gray-100 w-96 " action="post" @submit.prevent="proceed">
      <div>
        <label class="text-sm underline block" for="fir">Login:</label>
        <input id="fir" v-model="fir" class="mb-4 w-full focus:shadow-inner transition shadow-xs focus:outline-none border border-gray-200 text-gray-600 px-3 py-2 rounded-md flex-grow " type="text" name="fir">
      </div>
      <div>
        <label class="text-sm underline block" for="sec">Password:</label>
        <input id="sec" v-model="sec" class="mb-4 w-full focus:shadow-inner transition shadow-xs focus:outline-none border border-gray-200 text-gray-600 px-3 py-2 rounded-md flex-grow " type="password" name="sec">
      </div>
      <div>
        <button class=" hover:bg-indigo-500 text-gray-600 hover:text-white transition px-2 py-2 border border-gray-200 rounded-md hover:shadow-sm">
          Sign Up
        </button>
      </div>
    </form>
  </div>
</template>

<script>
export default {
  data: () => ({
    fir: '',
    sec: ''
  }),
  methods: {
    proceed () {
      try {
        this.$fireauth.signup(this.fir, this.sec)
      } catch (error) {
        // TODO: remove console
        console.error(error) // eslint-disable-line
      }
    }
  }
}
</script>

<style>

</style>
