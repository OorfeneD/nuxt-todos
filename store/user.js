export const state = () => {
  return {
    user: null
  }
}

export const getters = {
  user: state => state.user
}

export const mutations = {
  setUser (state, data) {
    state.user = data
  }
}

export const actions = {
  setUser ({ state, commit }, user) {
    if (state.user) {
      // TODO: remove console
      console.error('User is already logged in') // eslint-disable-line
    } else {
      commit('setUser', user)
    }
  },
  resetUser ({ commit }) {
    commit('setUser', null)
  }
}
