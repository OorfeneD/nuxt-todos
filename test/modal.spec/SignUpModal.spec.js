import { shallowMount } from '@vue/test-utils'
import { login, password } from '../mocks/user.mock'
import SignUpModal from '@/components/modal/SignUpModal'

describe('SignUpModal', () => {
  let wrapper = shallowMount(SignUpModal)
  beforeEach(() => {
    wrapper = shallowMount(SignUpModal)
  })
  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })
  test('matchs snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('emits event on from submit', () => {
    wrapper.find('input#fir').setValue(login)
    wrapper.find('input#sec').setValue(password)
    wrapper.find('form').trigger('submit')
    expect(wrapper.emitted().submit).toBeTruthy()
  })
  test('emitted data has login and pass', () => {
    wrapper.find('input#fir').setValue(login)
    wrapper.find('input#sec').setValue(password)
    wrapper.find('form').trigger('submit')
    expect(wrapper.emitted().submit[0]).not.toBeUndefined()
    expect(wrapper.emitted().submit[0][0]).not.toBeUndefined()
    expect(wrapper.emitted().submit[0][0].data).not.toBeUndefined()
    expect(wrapper.emitted().submit[0][0].data).toMatchObject([login, password])
  })
})
