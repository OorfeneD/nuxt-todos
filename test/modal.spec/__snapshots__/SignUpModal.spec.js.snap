// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpModal matchs snapshot 1`] = `
"<div class=\\"p-4 bg-white rounded-xl shadow-md my-5 border border-gray-100 w-96 \\">
  <header class=\\"flex justify-between border-b border-gray-200 pb-2 mb-4\\">
    <p><b>Sign up</b> with e-mail and password</p> <button>
      <icon close=\\"\\"></icon>
    </button>
  </header>
  <form action=\\"post\\">
    <div><label for=\\"fir\\" class=\\"text-sm underline block\\">E-mail:</label> <input id=\\"fir\\" type=\\"text\\" name=\\"fir\\" class=\\"mb-4 w-full focus:shadow-inner transition shadow-xs focus:outline-none border border-gray-200 text-gray-600 px-3 py-2 rounded-md flex-grow \\"></div>
    <div><label for=\\"sec\\" class=\\"text-sm underline block\\">Password:</label> <input id=\\"sec\\" type=\\"password\\" name=\\"sec\\" class=\\"mb-4 w-full focus:shadow-inner transition shadow-xs focus:outline-none border border-gray-200 text-gray-600 px-3 py-2 rounded-md flex-grow \\"></div>
    <div><button class=\\" hover:bg-indigo-500 text-gray-600 hover:text-white transition px-2 py-2 border border-gray-200 rounded-md hover:shadow-sm\\">
        Sign Up
      </button></div>
  </form>
</div>"
`;
